openvdb/unittest/TestFastSweeping.cc

```diff
@@ -0,0 +1,635 @@
+        const std::string message = diagnose.check(test,
+                                                   false,// don't generate a mask grid
+                                                   true,// check active voxels
+                                                   false,// ignore active tiles since a level set has none
+                                                   false);// no need to check the background value
+        CPPUNIT_ASSERT(message.empty());
+        CPPUNIT_ASSERT_EQUAL(Index64(0), diagnose.failureCount());
+        //std::cout << "\nOutput 1: " << message << std::endl;
+    }
+    {// Make sure all active voxels fail the following test
+        tools::Diagnose<FloatGrid> diagnose(*grid2);
+        tools::CheckNormGrad<FloatGrid> test(*grid2, std::numeric_limits<float>::min(), 0.99f);
+        const std::string message = diagnose.check(test,
+                                                   false,// don't generate a mask grid
+                                                   true,// check active voxels
+                                                   false,// ignore active tiles since a level set has none
+                                                   false);// no need to check the background value
+        CPPUNIT_ASSERT(!message.empty());
+        CPPUNIT_ASSERT_EQUAL(voxelCount, diagnose.failureCount());
+        //std::cout << "\nOutput 2: " << message << std::endl;
+    }
+    {// Make sure all active voxels fail the following test
+        tools::Diagnose<FloatGrid> diagnose(*grid2);
+        tools::CheckNormGrad<FloatGrid> test(*grid2, 1.01f, std::numeric_limits<float>::max());
+        const std::string message = diagnose.check(test,
+                                                   false,// don't generate a mask grid
+                                                   true,// check active voxels
+                                                   false,// ignore active tiles since a level set has none
+                                                   false);// no need to check the background value
+        CPPUNIT_ASSERT(!message.empty());
+        CPPUNIT_ASSERT_EQUAL(voxelCount, diagnose.failureCount());
+        //std::cout << "\nOutput 3: " << message << std::endl;
+    }
+}// dilateSignedDistance
+
+
+void
+TestFastSweeping::testMaskSdf()
+{
+    using namespace openvdb;
+    // Define parameterS FOR the level set sphere to be re-normalized
+    const float radius = 200.0f;
+    const Vec3f center(0.0f, 0.0f, 0.0f);
+    const float voxelSize = 1.0f, width = 3.0f;//half width
+    const float new_width = 50;
+
+    // Define a simple lambda function to write a grid to disk
+    // This is useful for visual inspections in Houdini!
+    auto writeFile = [](std::string name, FloatGrid::Ptr grid){
+        io::File file(name);
+        file.setCompression(io::COMPRESS_NONE);
+        GridPtrVec grids;
+        grids.push_back(grid);
+        file.write(grids);
+    };
```

Comment:
lots of writing in these unit tests

Reply:
yes not ideal but it was the easiest way to validate the output. I can remove the writes

Reply:
removed them

Reply:
I think I can still see a few uses of the `writeFile` lambda being invoked in this file?